tests/loganalyzer/loganalyzer.py

```diff
@@ -0,0 +1,232 @@
+import sys
+import logging
+import os
+import re
+import time
+import pprint
+import system_msg_handler
+
+from system_msg_handler import AnsibleLogAnalyzer as ansible_loganalyzer
+from os.path import join, split
+from os.path import normpath
+
+ANSIBLE_LOGANALYZER_MODULE = system_msg_handler.__file__.replace(r".pyc", ".py")
+COMMON_MATCH = join(split(__file__)[0], "loganalyzer_common_match.txt")
+COMMON_IGNORE = join(split(__file__)[0], "loganalyzer_common_ignore.txt")
+COMMON_EXPECT = join(split(__file__)[0], "loganalyzer_common_expect.txt")
+SYSLOG_TMP_FOLDER = "/tmp/pytest-run/syslog"
+
+
+class LogAnalyzerError(Exception):
+    """Raised when loganalyzer found matches during analysis phase."""
+    def __repr__(self):
+        return pprint.pformat(self.message)
+
+
+class LogAnalyzer:
+    def __init__(self, ansible_host, marker_prefix, dut_run_dir="/tmp"):
+        self.ansible_host = ansible_host
+        self.dut_run_dir = dut_run_dir
+        self.extracted_syslog = os.path.join(self.dut_run_dir, "syslog")
+        self.marker_prefix = marker_prefix
+        self.ansible_loganalyzer = ansible_loganalyzer(self.marker_prefix, False)
+
+        self.match_regex = []
+        self.expect_regex = []
+        self.ignore_regex = []
+        self._markers = []
+
+    def _add_end_marker(self, marker):
+        """
+        @summary: Add stop marker into syslog on the DUT.
+
+        @return: True for successfull execution False otherwise
+        """
+        self.ansible_host.copy(src=ANSIBLE_LOGANALYZER_MODULE, dest=os.path.join(self.dut_run_dir, "loganalyzer.py"))
+
+        cmd = "python {run_dir}/loganalyzer.py --action add_end_marker --run_id {marker}".format(run_dir=self.dut_run_dir, marker=marker)
+
+        logging.debug("Adding end marker '{}'".format(marker))
+        self.ansible_host.command(cmd)
+
+    def __enter__(self):
+        """
+        Store start markers which are used in analyze phase.
+        """
+        self._markers.append(self.init())
+
+    def __exit__(self, *args):
+        """
+        Analyze syslog messages.
+        """
+        self.analyze(self._markers.pop())
+
+    def _verify_log(self, result):
+        """
+        Verify that total match and expected missing match equals to zero or raise exception otherwise.
+        Verify that expected_match is not equal to zero when there is configured expected regexp in self.expect_regex list
+        """
+        if not result:
+            raise LogAnalyzerError("Log analyzer failed - no result.")
+        if result["total"]["match"] != 0 or result["total"]["expected_missing_match"] != 0:
+            raise LogAnalyzerError(result)
+
+        # Check for negative case
+        if self.expect_regex and result["total"]["expected_match"] == 0:
+            raise LogAnalyzerError(result)
+
+    def update_marker_prefix(self, marker_prefix):
+        """
+        @summary: Update configured marker prefix
+        """
+        self.marker_prefix = marker_prefix

```

Comment:
Actually it will be used every time when loganalyzer call "init" method.
By default this prefix will be test case name which is configured by "loganalyzer" pytest fixture.

Reply:
I was referring to the update_marker() method and not the marker_prefix var.

Reply:
Maybe I don't understand you correctly, but I don't think that it worth to highlight that "update_marker_prefix()" function will be used with "run_cmd()". It can create confusion. Because after call "update_marker_prefix()" function any farther loganalyzer calls will use updated marker, not only "run_cmd()".
Thoughts?

Reply:
I only see an assignment statement in the update_marker_prefix() method. As far as I understand, you need to call init() to actually write the new marker to the log file. Where is that done? I thought the use case for this comes with the run_cmd() method which actually calls init() and then analyse().

Reply:
Below there is general examples how/when init method can be called (it describes not only test file calls).
1. Use case
```
loganalyzer.update_marker_prefix("test_bgp")
# Add start marker to the DUT syslog
marker = loganalyzer.init()

# PERFORM BGP TEST CASE STEPS ...

loganalyzer.analyze(marker)
```

2. Use case
```
loganalyzer.update_marker_prefix("any_specific_prefix")
callback_result = loganalyzer.run_cmd(callback, ans_host)
```

Reply:
Makes sense. Can you add Use case 1  to the Readme? Use case 2 is mentioned there

Reply:
Those cases are already described in README.md. Please take a look at:
# Example 1
# Example 4